urllib3/connectionpool.py

```diff
@@ -790,11 +790,13 @@ def _prepare_proxy(self, conn):
             set_tunnel = conn.set_tunnel
         except AttributeError:  # Platform-specific: Python 2.6
             set_tunnel = conn._set_tunnel
-
+        host = self.host
+        if not (host.startswith('[') and host.endswith(']')):
+            host = '[' + host + ']'
```

Comment:
Hrrrrm. Iiiinteresting. I'm a bit inclined to actually want to apply some of the logic in the `__init__` where the square brackets are removed, to save a copy prior to that transformation for use when proxying. Maybe add a line that looks like `self._proxy_host = host.lower()`. How does that sound?